components/automate-ui/src/app/services/node-details/node-norunid-resolver.service.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { TestBed } from '@angular/core/testing';
+
+import { NodeNoRunIdResolverService } from './node-norunid-resolver.service';
+import { NodeRunsService } from './node-runs.service';
+import { Router } from '@angular/router';
+import { NodeRun } from '../../types/types';
+
+class MockRouter {
+  constructor() {}
+  navigate(_route: Array<string>) {
+  }
+}
+
+class MockNodeRunsService {
+  response: Promise<NodeRun> = Promise.resolve<NodeRun>(null);
+
+  constructor() {}
+
+  public getNodeRunsByID(_nodeId: string) {
+    return this.response;
+  }
+
+  setResponse(response: Promise<NodeRun>) {
+    this.response = response;
+  }
+}
+
+class MockRoute {
+  paramMap = {
+    get() {}
+  };
+}
+
+describe('NodeDetailsResolverService', () => {
+  let route;
+  let service: NodeNoRunIdResolverService;
+  const router = new MockRouter();
+  let nodeRunsService: MockNodeRunsService;
+
+  nodeRunsService = new MockNodeRunsService();
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        NodeNoRunIdResolverService,
+        { provide: NodeRunsService, useValue: nodeRunsService },
+        { provide: Router, useValue: router }]
+    });
+
+    service = TestBed.inject(NodeNoRunIdResolverService);
+    route = new MockRoute();
+  });
+
+  describe('null nodeRun is returned', () => {
+    beforeEach(() => {
+      nodeRunsService.setResponse(Promise.resolve<NodeRun>(null));
+    });
+
+    it('should redirect to missing runs page', (done) => {
```

Comment:
We do not really need a `describe` block containing a single `it` here. For one thing, it breaks up the test name in two pieces so it is less clear what is going on. Also, it requires more code. Consider this more direct form:
```suggestion
  it('redirects to missing runs page when null nodeRun is returned', () => {
    nodeRunsService.setResponse(Promise.resolve<NodeRun>(null));
```
Same comment applies to the remaining tests in this file--just a single `it` for each would suffice with something like these test names:
* redirects to client runs page when non-404 error is returned
* redirects to missing runs page when 404 error is returned